pyproject.toml

```diff
@@ -0,0 +1,52 @@
+[tool.poetry]
+name = "meshtastic"
+version = "2.3.11"
+description = "Python API & client shell for talking to Meshtastic devices"
+authors = ["Meshtastic Developers <contact@meshtastic.org>"]
+license = "GPL-3.0-only"
+readme = "README.md"
+
+[tool.poetry.dependencies]
+python = "^3.8,<3.13" # was 3.7 for production but, 3.8 is needed for pytap2, 3.9 is needed for pandas, bleak requires a max of 3.13 for some reason
+pyserial = "^3.5"
+protobuf = ">=5.26.0"
+dotmap = "^1.3.30"
+pexpect = "^4.9.0"
+pyqrcode = "^1.2.1"
+tabulate = "^0.9.0"
+timeago = "^1.0.16"
+webencodings = "^0.5.1"
+requests = "^2.31.0"
+pyparsing = "^3.1.2"
+pyyaml = "^6.0.1"
+pypubsub = "^4.0.3"
+bleak = "^0.21.1"
+mypy = "^1.10.0"
```

Comment:
Sorry, I know nothing about Poetry but Is it possible to separate out the things that are needed for testing vs. runtime?

Reply:
yes - there is a separate dev depends.  bleak and pypubsub need to stay in the runtime depends.  But good point mypy can move to the dev depends.  I'm not sure about pyyaml - haven't looked to see what that's used for.

Reply:
heh!   I just grepped and I can't find anything in meshtastic even using pyyaml.  I'll look at the history to see when it was added.

Reply:
pyyaml is used for some sort of save as yaml feature added sometime ago - so it needs to stay in runtime deps.